.github/workflows/inspektor-gadget.yml

```diff
@@ -988,50 +988,37 @@ jobs:
             context: "tools/dnstester"
             dockerfile: "tools/dnstester/Dockerfile"
             platform: "linux/amd64,linux/arm64"
-            filter-patterns:
-              - "tools/dnstester/*"
+            key-files: "tools/dnstester/*"
           - name: "gadget-builder"
             context: "/home/runner/work/inspektor-gadget/inspektor-gadget"
             dockerfile: "Dockerfiles/gadget-builder.Dockerfile"
             platform: "linux/amd64,linux/arm64"
-            filter-patterns:
-              - "include/**"
-              - "Dockerfiles/gadget-builder.Dockerfile"
-              - "cmd/common/image/Makefile.build"
+            key-files: "'include/**','Dockerfiles/gadget-builder.Dockerfile','cmd/common/image/Makefile.build'"
     steps:
     - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-    - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
-      id: filter
-      with:
-        # https://github.com/rhysd/actionlint/blob/main/docs/checks.md#check-type-check-expression
-        filters: |
-          pattern: ${{ toJson(matrix.image.filter-patterns) }}
-    - name: Check if we should build helpers
-      id: check-build-helpers
-      # always build the images on release, merge to main or to a release branch
-      if: steps.filter.outputs.pattern == 'true' || startsWith(github.ref_name, 'v') || startsWith(github.ref_name, 'release-') || github.ref_name == 'main'
-      run: |
-          echo "build=true" >> $GITHUB_OUTPUT
     - name: Set up Docker Buildx
-      if: steps.check-build-helpers.outputs.build == 'true'
       uses: docker/setup-buildx-action@b5ca514318bd6ebac0fb2aedd5d36ec1b5c232a2 # v3.10.0
+    - name: Cache Docker layers
+      uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-docker-${{ matrix.image.name }}-${{ hashFiles(matrix.image.key-files) }}
```

Comment:
Caching layers would mean things like `apt-get update` won't be using the latest changes so perhaps we should expire the cache? Having a date in the key should do the tick: https://github.com/marketplace/actions/cache#creating-a-cache-key

Reply:
> Caching layers would mean things like apt-get update won't be using the latest changes so perhaps we should expire the cache? 

What's the issue of not using the latest changes? Having old code with CVEs and that stuff?

> Having a date in the key should do the tick: https://github.com/marketplace/actions/cache#creating-a-cache-key

If we invalidate the cache too often, then what's the advantage of this PR?
Perhaps we can manually delete the cache and rerun this job when needed?